editoast/src/models/timetable.rs

```diff
@@ -212,9 +212,15 @@ pub async fn get_timetable_train_schedules_with_simulations(
     let simulation_outputs = SimulationOutput::belonging_to(&train_schedules)
         .load::<SimulationOutput>(&mut conn)
         .await?;
-    let result = train_schedules
+    let result: Vec<(TrainSchedule, SimulationOutput)> = train_schedules
         .into_iter()
-        .zip(simulation_outputs)
+        .flat_map(|schedule| {
+            let matching_output = simulation_outputs
+                .iter()
+                .find(|output| output.train_schedule_id == schedule.id);
```

Comment:
We could improve the complexity from $\mathcal{O}(n^2)$ to $\mathcal{O}(n)$. We could create a hashmap to avoid iterate on the `simulation_outputs` for each train schedule.